samples/apps/copilot-chat-app/copilotchat/Extensions/ServiceExtensions.cs

```diff
@@ -0,0 +1,183 @@
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Reflection;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using SemanticKernel.CopilotChat.Config;
+using SemanticKernel.CopilotChat.Models;
+using SemanticKernel.CopilotChat.Storage;
+
+namespace SemanticKernel.CopilotChat.Extensions;
+
+/// <summary>
+/// Extension methods for <see cref="IServiceCollection"/>.
+/// Add options and services for Copilot Chat.
+/// </summary>
+public static class CopilotChatServiceExtensions
+{
+    /// <summary>
+    /// Parse configuration into options.
+    /// </summary>
+    public static IServiceCollection AddCopilotChatOptions(this IServiceCollection services, ConfigurationManager configuration)
+    {
+        // AI service configurations for Copilot Chat.
+        // They are using the same configuration section as Semantic Kernel.
+        services.AddOptions<CopilotChatAIServiceOptions>(CopilotChatAIServiceOptions.CompletionPropertyName)
+            .Bind(configuration.GetSection(CopilotChatAIServiceOptions.CompletionPropertyName))
+            .ValidateOnStart()
+            .PostConfigure(TrimStringProperties);
+
+        services.AddOptions<CopilotChatAIServiceOptions>(CopilotChatAIServiceOptions.EmbeddingPropertyName)
+            .Bind(configuration.GetSection(CopilotChatAIServiceOptions.EmbeddingPropertyName))
+            .ValidateOnStart()
+            .PostConfigure(TrimStringProperties);
+
+        // Chat log storage configuration
+        services.AddOptions<ChatStoreOptions>()
+            .Bind(configuration.GetSection(ChatStoreOptions.PropertyName))
+            .ValidateOnStart()
+            .PostConfigure(TrimStringProperties);
+
+        // Azure speech token configuration
+        services.AddOptions<AzureSpeechOptions>()
+            .Bind(configuration.GetSection(AzureSpeechOptions.PropertyName))
+            .ValidateOnStart()
+            .PostConfigure(TrimStringProperties);
+
+        // Bot schema configuration
+        services.AddOptions<BotSchemaOptions>()
+            .Bind(configuration.GetSection(BotSchemaOptions.PropertyName))
+            .ValidateOnStart()
+            .PostConfigure(TrimStringProperties);
+
+        // Document memory options
+        services.AddOptions<DocumentMemoryOptions>()
+            .Bind(configuration.GetSection(DocumentMemoryOptions.PropertyName))
+            .ValidateOnStart()
+            .PostConfigure(TrimStringProperties);
+
+        // Planner options
+        services.AddOptions<PlannerOptions>()
+            .Bind(configuration.GetSection(PlannerOptions.PropertyName))
+            .ValidateOnStart()
+            .PostConfigure(TrimStringProperties);
+
+        services.AddOptions<PromptsOptions>()
+            .Bind(configuration.GetSection(PromptsOptions.PropertyName))
+            .ValidateOnStart()
+            .PostConfigure(TrimStringProperties);
+
+
```

Comment:
Limit that to one empty lines, not 3. I'm surprised our our formatter doesn't complain about stuff like this.